homeassistant/components/home_connect/button.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from .entity import HomeConnectEntity
+from . import _get_appliance_by_device_id
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+
+from .const import DOMAIN
+from homeassistant.const import CONF_DEVICE
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    BSH_PAUSE,
+    BSH_RESUME,
+)
+
+_PAUSE_KEY = "pause"
+_RESUME_KEY = "resume"
+_STOP_KEY = "stop"
+
+
+async def async_service_pause_program(hass: HomeAssistant, device_id: str):
+    """Service for pausing a program."""
+    await async_service_command(hass, device_id, BSH_PAUSE)
+
+
+async def async_service_resume_program(hass: HomeAssistant, device_id: str):
+    """Service for resuming a paused program."""
+    await async_service_command(hass, device_id, BSH_RESUME)
+
+
+async def async_service_stop_program(hass: HomeAssistant, device_id: str):
+    """Execute calls to services executing a stop of active program."""
+
+    appliance = _get_appliance_by_device_id(hass, device_id)
+    await hass.async_add_executor_job(appliance.stop_program)
+
+
+async def async_service_command(hass: HomeAssistant, device_id: str, command):
+    """Execute calls to services executing a command."""
+    appliance = _get_appliance_by_device_id(hass, device_id)
+    await hass.async_add_executor_job(appliance.execute_command, command)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up HomeConnect control buttons."""
+
+    hc_api = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[HomeConnectButton] = []
+
+    for device_dict in hc_api.devices:
+        entities.append(HomeConnectButton(device_dict[CONF_DEVICE], ButtonEntityDescription(key=_PAUSE_KEY, name="Pause")))
+        entities.append(HomeConnectButton(device_dict[CONF_DEVICE], ButtonEntityDescription(key=_RESUME_KEY, name="Resume")))
+        entities.append(HomeConnectButton(device_dict[CONF_DEVICE], ButtonEntityDescription(key=_STOP_KEY, name="Stop")))       
+                                        
+    async_add_entities(entities, True)
+
+
+class HomeConnectButton(HomeConnectEntity, ButtonEntity):
+    """HomeConnect Button Device."""
+
+    def __init__(
+        self, description: ButtonEntityDescription, device: HomeConnectEntity
+    ) -> None:
+        """Initialize the button."""
+        super().__init__(device, "Button")
+        self.entity_description = description
+        self._attr_unique_id = f"{super().unique_id}{description.key}"
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        if self.entity_description.key == _PAUSE_KEY:
+            await async_service_pause_program(self.hass, self.device.appliance.haId)
```

Comment:
You could expand the entity descriptions and put these functions there so instead you would have `await self.entity_description.button_fn(self.hass, self.device.appliance.haId)` only here

Reply:
@gjohansson-ST tried it, but then I get follow error:


```
homeassistant/components/home_connect/button.py:27: error: Attributes without a default cannot follow attributes with one  [misc]
homeassistant/components/home_connect/button.py:44: error: "HomeConnectDevice" has no attribute "stop_program"  [attr-defined]
homeassistant/components/home_connect/button.py:50: error: "HomeConnectDevice" has no attribute "execute_command"  [attr-defined]
Found 3 errors in 1 file (checked 1 source file)
```

Reply:
Ok added the mixin. But don't get this error
```
homeassistant/components/home_connect/button.py:51: error: "HomeConnectDevice" has no attribute "stop_program"  [attr-defined]
homeassistant/components/home_connect/button.py:57: error: "HomeConnectDevice" has no attribute "execute_command"  [attr-defined]
```

This is the same like here:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/home_connect/__init__.py#L162

Reply:
@gjohansson-ST @DavidMStraub, can you tell me what's wrong? with this two lines?

Reply:
It looks like you're calling the method on `homeassistant.components.home_connect.api.HomeConnectDevice` rather than `homeconnect.api.HomeConnectAppliance`, which is stored in the `appliance` property of the latter.

Reply:
@DavidMStraub yes thats the issue I see. 
So then i can change it directly to the HomeConnectDevice from the button directly.

So this should work, right?
```
async def async_service_command(
    hass: HomeAssistant, device: HomeConnectDevice, command
):
    """Execute calls to services executing a command."""
    appliance = device.appliance
    await hass.async_add_executor_job(appliance.execute_command, command)
```